### DISCUSSION TITLE
Focus on new tab

### DISCUSSION THREAD

**Discussion by AndnixSH** [hardkoded/puppeteer-sharp#2378] [Q&A]
When the page opens a page in a new tab, the puppeteer lost the focus and can't find the elements. How do I make it focus on new tab after it's open?

**Comment by kblok** [comment]
Each `page` instance maps to a single tab in the browser. If you want to act on a new tab with need to wait for that new target and then use that new page instance.

```
var targetPromise = context.WaitForTargetAsync((target) => target.Url == "The url of the new tab");
// Some action that will open a new tab
var promiseTarget = await targetPromise;
var targetPage = await promiseTarget.PageAsync();
```

**Reply by AndnixSH** [reply]
Hmm it doesn't target new tab even I gave the excat url of the new tab, I'm getting timeout after 30 sec

**Reply by kblok** [reply]
Have you tried breaking into the Func you pass to ``WaitForTargetAync` and see what you are getting there?

**Reply by AndnixSH** [reply]
Yeah, it just hangs on `var promiseTarget = await targetPromise;` until timeout. I could never get it to work, so I ended up using `browser.PagesAsync();` for looping instead. Works like a charm

**Reply by kblok** [reply]
What I mean is add a breakpoint here `(target) => target.Url == "The url of the new tab"`
